controlled_access_terms.tokens.inc

```diff
@@ -0,0 +1,364 @@
+<?php
+
+/**
+ * @file
+ * Contains controlled_access_terms.tokens.inc.
+ *
+ * This file provides islandora tokens.
+ */
+
+use Drupal\taxonomy\Entity\Term;
+use Drupal\media\Entity\Media;
+use Drupal\file\Entity\File;
+use Drupal\controlled_access_terms\EDTFUtils;
+use Drupal\Core\Render\BubbleableMetadata;
+
+/**
+ * Implements hook_token_info().
+ */
+function controlled_access_terms_token_info() {
+  $type = [
+    'name' => t('Islandora Tokens'),
+    'description' => t('Tokens for Islandora objects.'),
+  ];
+  $node['title'] = [
+    'name' => t("Node Title"),
+    'description' => t("The node's title"),
+  ];
+  $node['agent_author'] = [
+    'name' => t("Author"),
+    'description' => t("The node's Author linked agents"),
+  ];
+  $node['agent_contributor'] = [
+    'name' => t("Contributors"),
+    'description' => t("The node's Contributor linked agents"),
+  ];
+  $node['agent_publisher'] = [
+    'name' => t("Publishers"),
+    'description' => t("The node's Publishers linked agents"),
+  ];
+  $node['agent_creators'] = [
+    'name' => t("Creators"),
+    'description' => t("The node's Creators linked agents"),
+  ];
+  $node['publication_date'] = [
+    'name' => t("Publication date"),
+    'description' => t('Show the "Date Created" into YYYY/MM/DD format (handles EDTF format)'),
+  ];
+  $node['pdf_url'] = [
```

Comment:
This fella ought to be moved into the islandora module since it does the whole 'media lookup by media_use term' dealio.

Reply:
This guy still needs to get moved over into the islandora module, or you put a conditional around it to check for the islandora module.  At least then it'll be a soft dependency instead of an undeclared hard dependency.

Reply:
Ok -- this token has now been moved over to the `islandora_tokens_1171` branch of the islandora module https://github.com/asulibraries/islandora/tree/islandora_tokens_1171 (as well as renaming the helper function from "controlled_access_terms_*" to lookup the media by mime type).